sigar_windows.go

```diff
@@ -295,28 +308,304 @@ func (self *DiskList) Get() error {
+	ExecutionState  uint16 // Requires SeDebugPrivilege, will not be present without this privilege
+	ParentProcessId uint32
+	Priority        uint32
+	CommandLine     string
+	CreationDate    time.Time
+
+	VirtualSize    uint64
+	WorkingSetSize uint64
+	PageFaults     uint32
+
+	UserModeTime   uint64
+	KernelModeTime uint64
+
+	ReadOperationCount  uint64
+	ReadTransferCount   uint64
+	WriteOperationCount uint64
+	WriteTransferCount  uint64
+}
+
+type WindowsRunState int
+
+const (
+	WindowsRunStateUnknown = WindowsRunState(iota)
+	WindowsRunStateOther
+	WindowsRunStateReady
+	WindowsRunStateRunning
+	WindowsRunStateBlocked
+	WindowsRunStateSuspendedBlocked
+	WindowsRunStateSuspendedReady
+	WindowsRunStateTerminated
+	WindowsRunStateStopped
+	WindowsRunStateGrowing
+)
+
+func convertWindowsRunState(state WindowsRunState) RunState {
+	// This mapping may not be exact, see "ExecutionState" at
+	// https://msdn.microsoft.com/en-us/library/aa387976(v=vs.85).aspx
+	switch WindowsRunState(state) {
+	case WindowsRunStateReady:
+	case WindowsRunStateRunning:
+		return RunStateRun
+	case WindowsRunStateBlocked:
+	case WindowsRunStateSuspendedBlocked:
+		return RunStateIdle
+	case WindowsRunStateTerminated:
+		return RunStateZombie
+	case WindowsRunStateStopped:
+		return RunStateStop
+	}
+	return RunStateUnknown
+}
+
+func (self *ProcessList) Get() error {
+	var procs []Win32_Process
+	query := wmi.CreateQuery(&procs, "")
+	err := wmiClient.Query(query, &procs)
+	if err != nil {
+		return err
+	}
+
+	processes := make([]Process, 0, len(procs))
+	for _, proc := range procs {
+		var process Process
+
+		// ProcState
+		process.ProcState.Name = proc.Name
+		process.ProcState.Pid = int(proc.ProcessId)
+		process.ProcState.Ppid = int(proc.ParentProcessId)
+		process.ProcState.Priority = int(proc.Priority)
+		process.ProcState.State = convertWindowsRunState(WindowsRunState(proc.ExecutionState))
+
+		// ProcIo
+		process.ProcIo.ReadBytes = proc.ReadTransferCount
+		process.ProcIo.ReadOps = proc.ReadOperationCount
+		process.ProcIo.WriteBytes = proc.WriteTransferCount
+		process.ProcIo.WriteOps = proc.WriteOperationCount
+
+		// ProcMem
+		process.ProcMem.Size = proc.VirtualSize
+		process.ProcMem.Resident = proc.WorkingSetSize
+		process.ProcMem.PageFaults = uint64(proc.PageFaults)
+
+		// ProcTime
+		if proc.UserModeTime != 0 {
+			// Time in user mode, in 100 nanosecond units. Divide by 10000 to get milliseconds
+			process.ProcTime.User = proc.UserModeTime / 10000
+		}
+		if proc.KernelModeTime != 0 {
+			// Time in kernel mode, in 100 nanosecond units. Divide by 10000 to get milliseconds
+			process.ProcTime.Sys = proc.KernelModeTime / 10000
+		}
+		process.ProcTime.Total = process.ProcTime.User + process.ProcTime.Sys
+		// Convert proc.CreationDate to millis
+		process.ProcTime.StartTime = uint64(proc.CreationDate.UnixNano() / (int64(time.Millisecond) / int64(time.Nanosecond)))
+
+		// ProcArgs
+		process.ProcArgs.List = strings.Split(proc.CommandLine, " ")
```

Comment:
Thinking it over, I just wrapped the existing commandline into a 1-element slice rather than split by space, to avoid giving the impression the arguments were parsed.